narwhal/primary/src/primary.rs

```diff
@@ -713,6 +748,21 @@ impl PrimaryReceiverHandler {
             DagError::HeaderRequiresQuorum(header.digest())
         );
 
+        // Verify any randomness signature present in the header.
+        for m in header.system_messages().iter() {
+            if let SystemMessage::RandomnessSignature(round, sig) = m {
```

Comment:
I thought about adding this check and then decided not to for the reason you state, that it overcomplicates the proposer. There are lots of ways that validators could casue each other to use extra resources, I don't think this one would rank near the top of the list. And it'd be social suicide for a validator to do so because it would be trivial to see who is deliberately performing such an attack and everybody slash them. This would be much more of a concern on a public interface but it's protected on primary-to-primary